File: med/python/Question1679.py
from collections import defaultdict
from typing import List


class Solution:
    """
    方法一：枚举右维护左+哈希表
    """

    def maxOperations(self, nums: List[int], k: int) -> int:
        ans = 0
        cnt = defaultdict(int)
        for num in nums:
            if k - num in cnt:
                cnt[k - num] -= 1
                if cnt[k - num] == 0:
                    del cnt[k - num]
                ans += 1
            else:
                cnt[num] += 1
        return ans


print(Solution().maxOperations([3, 1, 3, 4, 3], 6))

File: med/cpp/Question1679.cpp
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
    int maxOperations(vector<int> &nums, int k) {
        int ans = 0;
        unordered_map<int, int> m;
        for (int i = 0; i < nums.size(); i++) {
            if (m.find(k - nums[i]) != m.end() && m[k - nums[i]] > 0) {
                m[k - nums[i]] -= 1;
                ans++;
            } else {
                m[nums[i]] += 1;
            }
        }
        return ans;
    }
};